sdk/cognitiveservices/azure-cognitiveservices-formrecognizer/DESIGN_DOC.md

```diff
@@ -0,0 +1,436 @@
+
+class ExtractedWord:
+    text: str
+    bounding_box: List[float]
+    confidence: float
+```
+
+### Receipt Sample
+```python
+from azure.ai.formrecognizer import FormRecognizerClient
+
+client = FormRecognizerClient(endpoint, credential)
+
+receipt_image = "https://raw.githubusercontent.com/Azure-Samples/cognitive-services-REST-api-samples/master/curl/form-recognizer/contoso-allinone.jpg"
+poller = client.begin_extract_receipt(receipt_image)
+receipt = poller.result()
+result = receipt[0]
+
+print("Receipt contained the following values: ")
+print("ReceiptType: {}").format(result.receipt_type)
+print("MerchantName: {}").format(result.merchant_name)
+print("MerchantAddress: {}").format(result.merchant_address)
+print("MerchantPhoneNumber: {}").format(result.merchant_phone_number)
+print("TransactionDate: {}").format(result.transaction_date)
+print("TransactionTime: {}").format(result.transaction_time)
+for item in result.receipt_items:
+    print("Item Name: {}").format(item.name)
+    print("Item Quantity: {}").format(item.quantity)
+    print("Item Price: {}").format(item.total_price)
+print("Subtotal: {}").format(result.subtotal)
+print("Tax: {}").format(result.tax)
+print("Tip: {}").format(result.tip)
+print("Total: {}").format(result.total)
+```
+
+### Layout Models
+
+```python
+class ExtractedLayoutPage:
+    tables: List[ExtractedTable]
+    page_number: int
+
+class ExtractedTable: 
+    List[List[TableCell]]
+    row_count: int
+    column_count: int
+
+class TableCell:
+    text: str
+    column_index: int
+    column_span: int
+    confidence: float
+    is_footer: bool
+    is_header: bool
+    row_index: int
+    row_span: int
+    bounding_box: List[float]
+    raw_field: List[ExtractedLine]
+
+class ExtractedLine:
+    text: str
+    bounding_box: List[float]
+    language: str
+    words: List[ExtractedWord]
+
+class ExtractedWord:
+    text: str
+    bounding_box: List[float]
+    confidence: float
+```
+
+### Layout Sample
+
+```python
+import pandas as pd
+from azure.ai.formrecognizer import FormRecognizerClient
+
+client = FormRecognizerClient(endpoint, credential)
+
+table_image = "https://i.stack.imgur.com/1FyIg.png"
+poller = client.begin_extract_layout(table_image)
+result = poller.result()
+
+dftable = pd.DataFrame(result[0].tables[0]) # page 1, table 1
+print(dftable)
+```
+
+## Custom
+
+The `CustomFormClient` provides all the operations necessary for training a custom model, analyzing a form with a 
+custom model, and managing a user's custom models on their account.
+
+The user can choose to train with or without labels using the methods `begin_labeled_training` or `begin_training`. 
+Both methods take as input a blob SAS uri or valid path to the documents to use for training. Each training method 
+will return a poller object which is used to get the training result.
+
+A custom model can be used to analyze forms using the `begin_extract_form` or `begin_extract_labeled_fields` methods.
```

Comment:
The "with labels" model will also extract tables. Is `begin_extract_labeled_fields` too specific?

Reply:
Good point. With the recent design changes of merging the result type, the custom form analysis methods will likely collapse into one `begin_extract_form` method. I'll update this design soon.